frontend/coprs_frontend/coprs/logic/complex_logic.py

```diff
@@ -798,13 +799,21 @@ def _delete_reason(cls, copr_chroots):
         # project owner. Or all of them for the same reason. Also all
         # architectures may be deleted by the project owner but on a different
         # day and therefore the remaining time may be different)
+        # See https://docs.pagure.org/copr.copr/developer_documentation/eol-logic.html
         reasons = {}
         for chroot in delete_chroots:
             reason_format = "{0} and will remain available for another {1} days"
-            reason = reason_format.format(
-                "disabled by a project owner" if chroot.is_active else "EOL",
-                chroot.delete_after_days,
-            )
+            reason = "EOL"
+            if chroot.is_active:
+                if chroot.deleted:
+                    # This is in `preserved` state, not `deleted`, otherwise we
+                    # wouldn't even ask for the _delete_reason().
+                    reason = "disabled by a project owner"
+                elif chroot.mock_chroot.rolling:
+                    reason = "a rolling chroot inactive for a long time"
+                else:
+                    raise exceptions.BadRequest(f"Unknown EOL reason {chroot.name}")
```

Comment:
This will probably show a 500 error page, making it unable to see the "broken" project?
I don't know if we should rather log this and display a big warning saying, that Copr admins should be contacted?

Reply:
BadRequest should be 400. This is a situation that shouldn't happen, probably caused by some wrong request, but :shrug:  I think exception should be preferred, so we fix the caller.

Reply:
/me nods